File: scalang/src/main/scala/com/kapilkoju/autopos/kernel/api/CudApi.scala
package com.kapilkoju.autopos.kernel.api

import com.kapilkoju.autopos.kernel.domain.AuditableEntity
import com.kapilkoju.autopos.kernel.service.AuditableRepository

abstract class CudApi[T <: AuditableEntity](override protected val repo: AuditableRepository[T],
                                            override protected val entityName: String,
                                            override protected val baseUrl: String)
  extends Api[T](repo, entityName, baseUrl)
    with CreateApi[T]
    with UpdateApi[T]
    with DeleteApi[T]

File: kotlang/src/main/java/com/kapilkoju/autopos/kernel/api/CudApi.kt
package com.kapilkoju.autopos.kernel.api

import com.kapilkoju.autopos.kernel.domain.AuditableEntity
import com.kapilkoju.autopos.kernel.service.AuditableRepository

abstract class CudApi<T : AuditableEntity>(override val repo: AuditableRepository<T>,
                                           override val entityName: String,
                                           override val baseUrl: String)
    : Api<T>, CreateApi<T>, UpdateApi<T>, DeleteApi<T>
